services/src/main/java/org/keycloak/userprofile/DeclarativeUserProfileProvider.java

```diff
@@ -151,16 +154,21 @@ protected UserProfileMetadata configureUserProfile(UserProfileMetadata metadata,
             return decoratedMetadata;
         }
 
-        ComponentModel model = getComponentModelOrCreate(session);
-        Map<UserProfileContext, UserProfileMetadata> metadataMap = model.getNote(PARSED_CONFIG_COMPONENT_KEY);
+        ComponentModel component = getComponentModel().orElse(null);
+
+        if (component == null) {
+            return DEFAULT_METADATA.computeIfAbsent(context, (c) -> decorateUserProfileForCache(decoratedMetadata, defaultRawConfig));
```

Comment:
This `computeIfAbsent` feeds data into a static final map `DEFAULT_METADATA`. The dependent data is from what I see dependent on the realm and user, and I don't see how much `DEFAULT_METADATA` can grow here, and how old entries are eventually cleared.

Reply:
@ahus1 I made most of these changes on top of @vramik work. You're right and I'll review this part. Not sure if the usages of ream there still make sense.